### DISCUSSION TITLE
save attachments of web service sender

### DISCUSSION THREAD

**Discussion by sbtnw** [nextgenhealthcare/connect#6128] [Q&A] [answered]
I have a Destination of "web service sender" type, that consumes a SOAP web service.
I can get its response using  responseMap.get('destination1').getMessage(); and parsing the fields with regex.
However, I can't access the attachments of the response (I can see them in soapUI). How can I do this, either trough MTOM or javascript writer?
I want to be able to save them do disk.
thanks

**Comment by pacmano1** [comment]
So you don't see the attachments at all?  Either in the attachments tab of the message viewer or the response itself?

**Reply by pacmano1** [reply]
Also, you can just call the SOAP endpoint via an http sender, I do that for the most part now due to limits on the web service sender.

**Reply by sbtnw** [reply]
do you have an example of that, please? (call the SOAP via http sender)
I have done it with a rest ws, not with a soap.
thanks

**Reply by pacmano1** [reply, marked as answer]
In the http sender outbound message template you would put the entire SOAP envelope suitable for your call.

In the headers you would have SOAPAction and the value would be whatever the SOAP action is.   Content type `text/xml,` data type text.   Content would be `${message.encodedData}`

You can mess with "Response Content" and "Parse Multipart" options to see what the result looks like.
